### PR TITLE
Extension implementation can inject texture loading strategy to ImporterContext

### PR DESCRIPTION
**Pull request by Santarh**
- fix namespace 
- define `ITextureDeserializer`
- `ImporterContext` can be injected `ITextureDeserializer`
- Remove NormalMap conversion on importing.

### CODE REVIEW

File: Assets/UniGLTF/Runtime/UniGLTF/IO/ImporterContext.cs [hunk 1]
```diff
@@ -37,14 +37,19 @@ public IAnimationImporter AnimationImporter
         public TextureFactory TextureFactory { get; }
         public MaterialFactory MaterialFactory { get; }
 
-        public ImporterContext(GltfParser parser, IReadOnlyDictionary<SubAssetKey, UnityEngine.Object> externalObjectMap = null)
+        public ImporterContext(
+            GltfParser parser,
+            IReadOnlyDictionary<SubAssetKey, UnityEngine.Object> externalObjectMap = null,
+            ITextureDeserializer textureDeserializer = null)
```
Comment (Santarh) [inline Assets/UniGLTF/Runtime/UniGLTF/IO/ImporterContext.cs:43]: 😄

File: Assets/VRMShaders/GLTF/IO/Runtime/ITextureDeserializer.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    public interface ITextureDeserializer
+    {
+        /// <summary>
+        /// imageData をもとに Texture2D を生成する
+        /// </summary>
+        /// <param name="imageData">データ</param>
+        /// <param name="useMipmap">Texture2D の mipmap が生成されるべきか否か</param>
+        /// <param name="colorSpace">Texture2D の色空間</param>
+        /// <returns></returns>
+        Task<Texture2D> LoadTextureAsync(byte[] imageData, bool useMipmap, ColorSpace colorSpace);
```
Comment (Santarh) [inline Assets/VRMShaders/GLTF/IO/Runtime/ITextureDeserializer.cs:19]: 😄

File: Assets/VRMShaders/GLTF/IO/Runtime/NormalConverter.cs [hunk 3]
```diff
@@ -19,28 +17,7 @@ private static Material Decoder
...
-                }
-                return m_encoder;
-            }
-        }
-
-        // GLTF data to Unity texture
-        // ConvertToNormalValueFromRawColorWhenCompressionIsRequired
-        public static Texture2D Import(Texture2D texture)
-        {
-            return TextureConverter.CopyTexture(texture, ColorSpace.Linear, false, Encoder);
```
Comment (Santarh) [inline Assets/VRMShaders/GLTF/IO/Runtime/NormalConverter.cs:39]: 😄

File: Assets/VRMShaders/GLTF/IO/Runtime/TextureFactory.cs [hunk 4]
```diff
@@ -99,14 +82,16 @@ public async Task<Texture> GetTextureAsync(TextureDescriptor texDesc)
...
-                    convertedTexture.SetSampler(texDesc.Sampler);
-                    m_textureCache.Add(subAssetKey, convertedTexture);
-                    DestroyResource(rawTexture);
-                    return convertedTexture;
+                    // no conversion. Unity's normal map is same with glTF's.
+                    //
+                    // > contrary to Unity’s usual convention of using Y as “up”
+                    // https://docs.unity3d.com/2018.4/Documentation/Manual/StandardShaderMaterialParameterNormalMap.html
+                    var data0 = await texDesc.Index0();
+                    var rawTexture = await _textureDeserializer.LoadTextureAsync(data0, texDesc.Sampler.EnableMipMap, ColorSpace.Linear);
```
Comment (ousttrue) [inline Assets/VRMShaders/GLTF/IO/Runtime/TextureFactory.cs:90]: _textureDeserializer.LoadTextureAsync

**Review by ousttrue — APPROVED** [review]
ITextureDeserializer

**Pull request merged**